testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java

```diff
@@ -30,15 +31,11 @@
 public class PlatformChecks
         extends ImageNameSubstitutor
 {
-    private static final boolean TESTCONTAINERS_SKIP_ARCH_CHECK = "true".equalsIgnoreCase(getenv("TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK"));
+    private static final boolean FAIL_ON_ARCHITECTURE_MISMATCH = "true".equalsIgnoreCase(firstNonNull(getenv("TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK"), "true"));
```

Comment:
I would like to keep check on by default

Reply:
This could be the only comment  of the review, so I wouldn't respond to "Please add a comemnt" first :) 

the check is causing issues, that's why i want to remove it.
For example, the direct trigger for this PR was when I was trying to run all iceberg tests locally , as https://github.com/trinodb/trino/pull/15441 affects many of them.
I cannot easily do this in Intellij (which tests require secrets to run?), so i switched to command line.
It's a pity when tests fail after 15 minutes because the metastore failed to load. 

I don't think the check is beneficial overall
i don't get at all, why the check is doing system exit (and there is no code comment explaining the reasoning)

i think we should not have the check enabled by default. But it's a cool check overall, we should run it on CI (which we don't do)

Reply:
The goal of the check was to save human time. Sometimes people do not notice they use emulation is in play and things seem to be working and later on they spend a lot of time on debugging why something breaks just to realize they used emulation. This happens when people work on something else (like planner which is very M1 friendly, so they even forget about what CPU they use) and then suddenly they start to work on sql-server.

This check was motivated because of such event and we wanted to avoid it in future.

> why the check is doing system exit

We should improve that. CC: @wendigo 

> I cannot easily do this in Intellij (which tests require secrets to run?)

Secrets should not be blocker to run tests locally. I think this part should get improved.

> tests fail after 15 minutes

Because of this check?

Maybe you should have `TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK` defined always or some maybe we need some other kill switch based on file for you? I know you know what you are doing, but I guess you might not be representative of all trino contributors  :)

Reply:
Why we should run it in the CI? We have full control over the architecture of the runners.

Reply:
> The goal of the check was to save human time.

noble goal, i agree with it

> Sometimes people do not notice they use emulation is in play and things seem to be working and later on they spend a lot of time on debugging why something breaks just to realize they used emulation.

That's a pity. I think Apple M1 is the main source of problems here.
However, engineers don't have choice. They cannot fix their setup and are unlikely to shop for a different laptop to run a test. So they disable the check and re-run... which takes time. does it still save human time?

> Secrets should not be blocker to run tests locally. I think this part should get improved.

Secrets are inherently required to run **all** tests locally. I don't think you can fix that  unless you can have Glue in docker or real S3 in docker (not "some S3-compatible system").

> > tests fail after 15 minutes
> Because of this check?

yep. depends on ordering of tests. the first class using a "bad" image wasn't run in the first 15 minutes of `mvn test -pl :trino-iceberg` for me

> Maybe you should have TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK defined always or some maybe we need some other kill switch based on file for you? I know you know what you are doing, but I guess you might not be representative of all trino contributors :)

I am sure I am not the only one having to workaround the check. See eg https://github.com/trinodb/trino/pull/16507#discussion_r1133057366.  If i cared about myself only, I wouldn't bother to open this PR.  It's not a time saver for me.

> Why we should run it in the CI? We have full control over the architecture of the runners.

Because we have no intention to use any form of docker emulation there.
Things may change in the future, but I believe that's the current state.